controllers/util/provider.go

```diff
@@ -77,3 +78,23 @@ func readAndDecodePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
 	}
 	return privateKey, nil
 }
+
+func DecodeOnboardingTicket(ticketString string) (*services.OnboardingTicket, error) {
+	ticketArr := strings.Split(ticketString, ".")
+	if len(ticketArr) != 2 {
+		return nil, fmt.Errorf("invalid ticket")
+	}
+
+	message, err := base64.StdEncoding.DecodeString(ticketArr[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode onboarding ticket: %v", err)
+	}
+
+	var ticketData services.OnboardingTicket
+	err = json.Unmarshal(message, &ticketData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal onboarding ticket %v", err)
+	}
```

Comment:
It seems decoding is already a part of the onboarding process here:
https://github.com/red-hat-storage/ocs-operator/blob/main/services/provider/server/server.go#L440-L454 which is the correct place to decode. 

I suggest renaming that method to `decodeAndValidateTicket` and return the decoded ticket as well

Reply:
Okay.